### PR TITLE
Rename `extra`

### PR DESCRIPTION
**Pull request by jolange**
Renamed `extra` to `worker_extra_args` and added a FutureWarning for `extra`.

If `worker_extra_args` is empty, `extra` will still be respected (see new tests).

Resolves https://github.com/dask/dask-jobqueue/issues/323

### CODE REVIEW

File: dask_jobqueue/jobqueue.yaml [hunk 1]
```diff
@@ -11,7 +11,8 @@ jobqueue:
     death-timeout: 60           # Number of seconds to wait if a worker can not find a scheduler
     local-directory: null       # Location of fast local storage like /scratch or $TMPDIR
     shared-temp-directory: null       # Shared directory currently used to dump temporary security objects for workers
-    extra: []
+    extra: null
```
Comment (jacobtomlinson) [inline dask_jobqueue/jobqueue.yaml:14]: Nit: Could you add a descriptive comment to each of these like the above config options?

Same for all below.

**Review by jacobtomlinson — APPROVED** [review]
This looks great many thanks! One small comment but otherwise looks good to me.

File: dask_jobqueue/jobqueue.yaml (hunk 1 above)
Comment (jolange) [reply dask_jobqueue/jobqueue.yaml]: Sure, done!

**Review by guillaumeeb — APPROVED** [review]
Perfect!

**Pull request merged**